tests/functional/raw-node/test/GCP/object/copy.js

```diff
@@ -0,0 +1,180 @@
+const assert = require('assert');
+const async = require('async');
+const { GCP } = require('../../../../../../lib/data/external/GCP');
+const { makeGcpRequest } = require('../../../utils/makeRequest');
+const { gcpRequestRetry } = require('../../../utils/gcpUtils');
+const { getRealAwsConfig } =
+    require('../../../../aws-node-sdk/test/support/awsConfig');
+
+const credentialOne = 'gcpbackend';
+const bucketName = `somebucket-${Date.now()}`;
+
+describe('GCP: COPY Object', function testSuite() {
+    this.timeout(180000);
+    const config = getRealAwsConfig(credentialOne);
+    const gcpClient = new GCP(config);
+
+    before(done => {
+        gcpRequestRetry({
+            method: 'PUT',
+            bucket: bucketName,
+            authCredentials: config.credentials,
+        }, 0, err => {
+            if (err) {
+                process.stdout.write(`err in creating bucket ${err}\n`);
+            }
+            return done(err);
+        });
+    });
+
+    after(done => {
+        gcpRequestRetry({
+            method: 'DELETE',
+            bucket: bucketName,
+            authCredentials: config.credentials,
+        }, 0, err => {
+            if (err) {
+                process.stdout.write(`err in creating bucket ${err}\n`);
+            }
+            return done(err);
+        });
+    });
+
+    describe('without existing object in bucket', () => {
+        it('should return 404 and \'Not Found\'', done => {
+            const missingObject = `nonexistingkey-${Date.now()}`;
+            const someKey = `somekey-${Date.now()}`;
+            gcpClient.copyObject({
+                Bucket: bucketName,
+                Key: someKey,
+                CopySource: `/${bucketName}/${missingObject}`,
```

Comment:
I guess a lot of my questions hinge on what the form of the `CopySource` param should take. Seems like it should not need a leading slash. Can you confirm?